presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -287,6 +290,34 @@ public List<UniqueConstraint<String>> getUniqueConstraints(MetastoreContext meta
         }
     }
 
+    @Override
+    public List<NotNullConstraint<String>> getNotNullConstraints(MetastoreContext metastoreContext, String dbName, String tableName)
+    {
+        try {
+            Optional<NotNullConstraintsResponse> notNullConstraintsResponse = retry()
+                    .stopOnIllegalExceptions()
+                    .run("getNotNullConstraints", stats.getGetNotNullConstraints().wrap(() ->
+                            getMetastoreClientThenCall(metastoreContext, client -> client.getNotNullConstraints("hive", dbName, tableName))));
+
+            if (!notNullConstraintsResponse.isPresent() || notNullConstraintsResponse.get().getNotNullConstraints().size() == 0) {
```

Comment:
as someone reading this for the first time, I find the original code far easier to follow than the rewritten "functional" code. I strongly prefer keeping the original.

Reply:
I think the readability argument is subjective to one's exposure with functional programming paradigms.

I am merely suggesting to use the idiomatic functional style with the `Optional`, because the point of having optional is to perform operations like this without having to write conditionals which tend to be subject to programmer-error. Using `Optional`'s API prevents relying on an IDEs advance syntax checking features to warn whether or not an `isPresent` check is performed before a `get`, etc. By using the API you guarantee that you never get a runtime exception due to a missing `Optional.isPresent` check.

Reply:
I'm afraid I disagree. The problem is not one's exposure to functional programming paradigms. It is the baroque design of Java's pseudo-functional APIs bolted on to the language years too late. Pretending Java is a functional language like Haskell or XSLT is a recipe for obfuscation.